torch/distributed/_functional_collectives.py

```diff
@@ -166,6 +166,38 @@ def broadcast(self: torch.Tensor, src: int, group: RANK_TYPES, tag: str = ""):
     return _maybe_wrap_tensor(tensor)
 
 
+def scatter(
+    self: torch.Tensor,
+    scatter_list: Optional[List[torch.Tensor]],
+    src: int,
+    group: RANK_TYPES,
+    tag: str = "",
+):
+    """
+    Scatters the list of data tensor to all processes in the given process group.
+
+    Args:
+        self (Tensor): dummy tensor of correct size to receive tensor.
+        scatter_list (List[Tensor]): list of tensors to scatter. Must be None on ranks 1+.
+        src (int): Source rank
+        group (ProcessGroup or List[int]): The process group to work on.
+        tag (str, optional): A unique identifier for the collective. Default: empty string
+    Returns:
+        Copy of dummy tensor with the broadcasted tensor.
+    """
+    if native_funcol_enabled():
+        group_name = _resolve_group_name(group, tag)
+        if scatter_list is None:
+            scatter_list = []
+        tensor = torch.ops._c10d_functional.scatter(self, scatter_list, src, group_name)
+    else:
+        tag, rankset, group_size = _expand_group(group, tag)
+        tensor = torch.ops.c10d_functional.scatter(
+            self, scatter_list, src, tag, rankset, group_size
+        )
```

Comment:
It would be great if we don't include this code path (and the python impl). I had to include them for other collectives because they were introduced before native funcol.